### PR TITLE
Changelog for v0.10.1 release

### PR DESCRIPTION
**Pull request by niftynei**
Notes for the release

Changelog-None.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,122 @@ All notable changes to this project will be documented in this file.
...
+ - JSON-RPC: `autocleaninvoice` now returns an object, not a raw string. ([#4501])
+ - JSON-RPC: `fundpsbt` will not include UTXOs that aren't economic (can't pay for their own fees), unless 'all' ([#4509])
+ - JSON-RPC: `close` now always returns notifications on delays. ([#4465])
+ - Protocol: All new invoices require a `payment_secret` (i.e. modern TLV format onion) ([#4646])
+ - Protocol: We can no longer connect to peers which don't support `payment_secret`. ([#4646])
+ - Protocol: We will now reestablish and negotiate mutual close on channels we've already closed (great if peer has lost their database). ([#4559])
+ - Protocol: We now assume nodes support TLV onions (non-legacy) unless we have a `node_announcement` which says they don't. ([#4646])
+ - Protocol: Use a more accurate fee for mutual close negotiation. ([#4619])
+ - Protocol: channel feerates reduced to bitcoind's "6 block ECONOMICAL" rate. ([#4507])
+ - keysend now uses 22 for the final CTLV, making it rust-lightning compatible. ([#4548])
```
Comment (vincenzopalazzo) [inline CHANGELOG.md:56]: Maybe rust-lightning it is out of the dictionary (if any exist)?

**Review by vincenzopalazzo — COMMENTED** [review]
LGTM https://github.com/ElementsProject/lightning/pull/4675/commits/2e419178429b3044afb21c14766676fc4c857ab9

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,122 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [v0.10.1rc1] - 2021-07-21: "<awaiting name selection>"
+
+This release named by @nalinbhardwaj.
+
+### Added
+
+ - JSON-RPC: `invoice` now outputs explicit `payment_secret` it its own field. ([#4646])
```
Comment (shesek) [inline CHANGELOG.md:13]: s/it its/as its/?

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,122 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [v0.10.1rc1] - 2021-07-21: "<awaiting name selection>"
+
+This release named by @nalinbhardwaj.
+
```
Comment (rustyrussell) [inline CHANGELOG.md:10]: *NOTE*: dual funding and offers protocols have changed, and are incompatible with older releases (they're both still draft)

**Review by rustyrussell — APPROVED** [review]
Ack 9ad704bc6edd4f5c1c8fd6100e125d5eed9894a2

Wow, this is big release!

**Comment by rustyrussell** [comment]
Also, need to note that rebalance and drain plugins will no longer work with this release due to payment_secret being compulsory, so they should be re-downloaded!

**Review by rustyrussell — APPROVED** [review]
Ack 50ffe6b34f9f023b61cdb783734949aba3ba5aeb

**Pull request merged**